packages/core/src/codewhisperer/service/securityScanHandler.ts

```diff
@@ -168,7 +225,26 @@ function getMd5(fileName: string) {
     return hasher.digest('base64')
 }
 
-export function throwIfCancelled() {
+export function throwIfCancelled(scanType: CodeWhispererConstants.SecurityScanType) {
+    switch (scanType) {
+        case CodeWhispererConstants.SecurityScanType.Project:
+            if (codeScanState.isCancelling()) {
+                throw new CodeScanStoppedError()
+            }
+            break
+        case CodeWhispererConstants.SecurityScanType.File:
+            if (!CodeScansState.instance.isScansEnabled()) {
+                throw new CodeScanStoppedError()
+            }
+            break
+        default:
+            getLogger().warn(`Unknown scan type: ${scanType}`)
+            break
+    }
+}
+
+// TODO: Remove this
```

Comment:
Are we removing all the old functions?

Reply:
Yes they will be removed in a future PR to this branch